src/d3a/d3a_core/redis_connections/aggregator_connection.py

```diff
@@ -39,7 +48,15 @@ def _add_batch_event(self, device_uuid, event, batch_event_dict):
 
         batch_event_dict[aggregator_uuid].append(event)
 
-    def add_batch_market_event(self, device_uuid, event):
+    def _add_grid_stats_to_market_event(self, device_uuid, global_objects):
+        market_info = default_market_info
```

Comment:
Unfortunately I think a deepcopy is necessary here, in order to generate a new instance of the default market info dict. The way it is implemented ATM only the reference is copied, and will buffer incrementally the value the more times this method is called.

Reply:
hach, yes. You just shared this this morning. Will do so, Thanks